OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -8,236 +8,237 @@
  */
 #endregion
 
-using System;
 using System.Collections.Generic;
-using System.Diagnostics;
-using System.Linq;
-using OpenRA;
-using OpenRA.Primitives;
-using OpenRA.Support;
+using OpenRA.Mods.Common.Pathfinder;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Traits
 {
+	using System;
+	using System.Diagnostics;
+	using System.Linq;
+
 	[Desc("Calculates routes for mobile units based on the A* search algorithm.", " Attach this to the world actor.")]
 	public class PathFinderInfo : ITraitInfo
 	{
-		public object Create(ActorInitializer init) { return new PathFinder(init.World); }
+		public object Create(ActorInitializer init)
+		{
+			return new PathFinderCacheDecorator(new PathFinder(init.World), new PathCacheStorage(init.World));
+		}
 	}
 
-	public class PathFinder
+	public interface IPathFinder
 	{
-		const int MaxPathAge = 50;	/* x 40ms ticks */
-		static readonly List<CPos> EmptyPath = new List<CPos>(0);
+		/// <summary>
+		/// Calculates a path for the actor from source to destination
+		/// </summary>
+		/// <param name="from">The start node</param>
```

Comment:
Style nit: These param descriptions don't give us any extra information, so our convention is to omit them.